backend/modules/auth/router.py

```
@@ -0,0 +1,654 @@
+    add_cors_headers(response)
+    
+    google_oauth_provider.state_store.delete_state(role_data.state)
+    
+    return google_oauth_provider.create_user_login_response(new_user)
+
+
+@router.post("/google/link", response_model=UserLoginResponse)
+async def link_google_account(
+    request: AccountLinkingRequest,
+    response: Response,
+    db: Session = Depends(get_db)
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)